External/SPMmodified/xASL/xASL_io_SaveNifti.m

```diff
@@ -60,20 +70,45 @@ function xASL_io_SaveNifti(pathOrigNifti, pathNewNifti, imNew, nBits, bGZip, cha
 	end
 end
 
+% JSON saving
+% Options are:
+% 1. bSaveJson     = false                       -> we don't save a new JSON sidecar
+% 2. bCopyOrigJson = true  & JsonFields = empty  -> we copy the original JSON sidecar only
+% 3. bCopyOrigJson = false & JsonFields = filled -> we create a new JSON sidecar with new fields only
+% 4. bCopyOrigJson = true  & JsonFields = filled -> we merge the original JSON sidecar with the new fields
+%                                                   the new fields have priority
+bSaveJson = false;
+
+if nargin < 7 || isempty(bCopyOrigJson)
+    bCopyOrigJson = false;
+else
+    bSaveJson = true;

```

Comment:
That won't work. If you provide bCopyOrigJson == false, it will locate it and set bSaveJson = true :)

Reply:
I don't get your comment. Are you sure?

Reply:
You set the input parameter bCopyOrigJson == false, and JsonFields == [].
So at line 82, it will detect that bCopyOrigJson was assigned, so it will just to line 84 and 85 and set bSaveJson = true.

So despite bCopyOrigJson == false, it will attempt to save the JSON.
But of course later on - it will not load the reference JSON and you provided empty fields, so it will save an empty JSON.

The solution is to change line 84 to elseif bCopyOrigJson

Reply:
Fixed in cd45719d06c30b9ccc2ef705ac80dc4e37562912